enhancements/single-node/single-node-openshift-day2-workers.md

```diff
@@ -0,0 +1,453 @@
+single control-plane node clusters which have worker nodes added to them.
+
+- Define how the installer may be modified to generate the worker ignition
+manifest, even when doing bootstrap-in-place "none"-platform single
+control-plane node cluster installation
+
+### Non-Goals
+
+- Deal with clusters that have less nodes than they had during day-1
+installation. The rationale for this being a non-goal is that it seems that
+removing the "initial" nodes of a cluster causes a lot of issues even in a
+regular, multi-node installation. So dealing with similar issues in single
+control-plane node clusters is also out of scope for this enhancement. For
+example, installing a cluster with 3 control-plane nodes and 3 workers results
+in unschedulable control-plane nodes. Scaling down the worker machine-sets from
+3 to 0 post-installation does not magically make the control-plane nodes
+schedulable, and as a result a lot of infrastructure workloads fail to schedule
+and operators degrade. User intervention is required to fix this.
+
+- Deal with expansion of the single-node control-plane by adding more control-plane nodes
+
+- Deal with expansion of clusters that have been installed before the
+implementation of this enhancement (if possible, their expansion may addressed
+of with documentation outlining the steps that the user has to take to enable
+expansion).
+
+- Adjust the baremetal platform to support single control-plane node
+installations. The baremetal platforms solves the "floating ingress" issue by
+using virtual IP addresses/keepalived.
+
+- Deal with users who want their ingress traffic to not go through the single
+control-plane node.
+
+- Deal with non-cloud, non-"none" platforms such as baremetal, vSphere, etc.
+
+## Proposal
+
+Set the "Infrastructure Topology" value to "SingleReplica" in almost all 
+circumstances where the "Control Plane Topology" is also "SingleReplica".
+
+Make sure the default `IngressController` CR created by the Cluster Ingress
+Operator targets the "master" pool rather than the "worker" pool whenever the
+"Infrastrcture Topology" is set to "SingleReplica". This will ensure that the
+`router-default` deployment created by the Cluster Ingress Operator will always
+run on the single control plane node, and as a result any
+`*.apps.<cluster>.<base>` DNS entries which originally pointed at the single
+control plane node will remain correct even in the face of newly added worker
+nodes.
+
+Make sure worker node ignition files are generated even in bootstrap-in-place
+single control-plane node "none"-platform installation.
+
+### User Stories
+
+- As an OpenShift cluster administrator, I want to install a single control-plane
+node cluster with additional worker nodes, so that my cluster will be able to
+handle larger computation demands.
+
+- As an OpenShift cluster administrator, I want to add worker nodes to my
+existing single control-plane node cluster, so that it'll be able to meet
+growing computation demands.
```

Comment:
I personally don't have answers to these questions, I suggest taking a look at MGMT-8414 and the documents linked in it. 

Some things that I think may make this make more sense -

- If my understanding of the motivations behind the requirements is correct, the most common use case is only a single additional worker being added, and in that case it's not enough to form a HA control plane. Situations with 2 or 3 more workers would be the exception.

- 3 machines behaving as 1 control plane + 2 workers are not equivalent to 3 machines that all act as control plane  + workers (compact cluster) - you pay for the HA of the control plane with, for example, a lot of memory for maintaining 3 k8s api servers rather than just 1. 

- I wouldn't be surprised if users actually *did* want to simply transform their 1 control plane + 2 workers cluster into a simple HA compact cluster if given the option to, but this is not something we support and I believe is a harder technical challenge to tackle - users would like to avoid re-installing their existing 1 + 2 cluster into a compact cluster - so for now, 1 + 2 is a good simpler middle-ground to answer user needs before we begin supporting live control plane expansion.